### PR TITLE
lua: expose vim.validate_command

### PR DESCRIPTION
**Pull request by teto**
well I thought it was exposed because it appeared in the help but it doesnt so here is the follow up of https://github.com/neovim/neovim/pull/11832

I haven't changed the function

### CODE REVIEW

**Comment by justinmk** [comment]
> because it appeared in the help

Because it wasn't marked private (i.e. underscore prefix).

File: runtime/lua/vim/lsp.lua [hunk 1]
```diff
@@ -121,28 +121,6 @@ local function validate_encoding(encoding)
       or error(string.format("Invalid offset encoding %q. Must be one of: 'utf-8', 'utf-16', 'utf-32'", encoding))
 end
 
-local function validate_command(input)
```
Comment (justinmk) [inline runtime/lua/vim/lsp.lua]: To make it accessible as `vim.lsp._...`:

    function lsp._cmd_parts(input)

The `lsp.` prefix makes it available. The underscore avoids "publishing" it in docs (and tells users this is not a public API).

Doesn't need to move to `shared.lua`.

**Comment by teto** [comment]
> This function is a code smell. Why are we specifying commands as strings instead of lists?

Thats something we may want to fix in the lsp code before 0.5 then but the change may break several configs.

**Comment by teto** [comment]
and if we forbid strings; the code becomes simpler so we dont need to expose it anymore.

**Comment by justinmk** [comment]
indeed, but meanwhile https://github.com/neovim/neovim/pull/11847#discussion_r377304041 unblocks you, right? Or else let's close this?

**Comment by teto** [comment]
> This function is a code smell. Why are we specifying commands as strings instead of lists?

It would be simpler to go straight to this but I don't want to `break` nvim-lsp without approval. I can update all `skeletons` (now `configs`) to achieve this.

**Comment by justinmk** [comment]
Not sure what you mean, nvim-lsp is explicitly HEAD-only as mentioned in the readme. Not worth spending time deliberating or discussing this. Let's either make the change I suggested or close this.

**Comment by teto** [comment]
ready. it now only accepts list. I checked nvim-lsp and everything is implemented as a list already so less headache than I anticipated.

**Comment by justinmk** [comment]
There's not much need for `lsp._cmd_parts()` if we restrict to a list. But harmless for now I guess.

**Pull request merged**